### PR TITLE
fix(FEC-10722): add KS to external captions

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

issue: ks didn't add in the correct place.
solution: set the ks as a parameter on the end of URL and if the URL ended with / add the ks with slashes

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: test/src/k-provider/ovp/be-data.js [hunk 1]
```diff
@@ -2937,7 +2938,7 @@ const EntryWithBumperWithKs = {
           webVttUrl:
             'http://cdntesting.qa.mkaltura.com/api_v3/index.php/service/caption_captionasset/action/serveWebVTT/captionAssetId/0_njhnv6na/segmentIndex/-1/version/2/captions.vtt?testParam=abc',
           url:
-            'http://cdntesting.qa.mkaltura.com/api_v3/index.php/service/caption_captionAsset/action/serve/captionAssetId/0_njhnv6na/v/2/?testParam=abc',
+            'http://cdntesting.qa.mkaltura.com/api_v3/index.php/service/caption_captionAsset/action/serve/captionAssetId/0_njhnv6na/v/2/captions.vtt?testParam=abc',
```
Comment (OrenMe) [inline test/src/k-provider/ovp/be-data.js:2941]: this is not correct, the url for srt doesn't have captions.vtt suffix, only the webVttUrl has it.
comment applies to all other samples you changed.

File: test/src/k-provider/ovp/be-data.js [hunk 2]
```diff
@@ -3410,7 +3411,7 @@ const EntryWithBumperWitNoSources = {
           language: 'Spanish',
           webVttUrl:
             'http://cdntesting.qa.mkaltura.com/api_v3/index.php/service/caption_captionasset/action/serveWebVTT/captionAssetId/0_njhnv6na/segmentIndex/-1/version/2/captions.vtt',
-          url: 'http://cdntesting.qa.mkaltura.com/api_v3/index.php/service/caption_captionAsset/action/serve/captionAssetId/0_njhnv6na/v/2',
+          url: 'http://cdntesting.qa.mkaltura.com/api_v3/index.php/service/caption_captionAsset/action/serve/captionAssetId/0_njhnv6na/v/2/',
```
Comment (OrenMe) [inline test/src/k-provider/ovp/be-data.js:3414]: this is not the way the url comes from BE, it doesn't come with `/` at the end

**Review by OrenMe — CHANGES_REQUESTED** [review]

File: test/src/k-provider/ovp/be-data.js (hunk 1 above)
Comment (Yuvalke) [reply test/src/k-provider/ovp/be-data.js:2941]: it should change to `.srt` instead `.vtt` it could come as `caption.srt`

File: test/src/k-provider/ovp/be-data.js (hunk 2 above)
Comment (OrenMe) [reply test/src/k-provider/ovp/be-data.js:3414]: are you saying from knowledge? afaics it is not - check with BE team

File: test/src/k-provider/ovp/media-config-data.js [hunk 3]
```diff
@@ -1106,15 +1106,15 @@ const EntryWithBumperWithKs = {
         language: 'en',
         label: 'En',
         url:
-          'http://cdntesting.qa.mkaltura.com/api_v3/index.php/service/caption_captionAsset/action/serve/captionAssetId/0_kozg4x1x/v/2/ks/YmMzNzUyZWM4ZmVkYjRiMzRlOTBlYTZjMGY2YTI1NzRkZDUwZjZjNnwxMDkxOzEwOTE7MTYwNTcyMjI5NDsyOzE2MDU2MzU4OTQuMTA0MzthdmkuYmFydWNoQGthbHR1cmEuY29tOyosZGlzYWJsZWVudGl0bGVtZW50Ozs'
+          'http://cdntesting.qa.mkaltura.com/api_v3/index.php/service/caption_captionAsset/action/serve/captionAssetId/0_kozg4x1x/v/2/captions.srt?ks=YmMzNzUyZWM4ZmVkYjRiMzRlOTBlYTZjMGY2YTI1NzRkZDUwZjZjNnwxMDkxOzEwOTE7MTYwNTcyMjI5NDsyOzE2MDU2MzU4OTQuMTA0MzthdmkuYmFydWNoQGthbHR1cmEuY29tOyosZGlzYWJsZWVudGl0bGVtZW50Ozs'
```
Comment (OrenMe) [inline test/src/k-provider/ovp/media-config-data.js:1109]: again, this is not the way I know BE returns the response today. Please verify with BE how the response urls come

Comment (Yuvalke) [reply test/src/k-provider/ovp/media-config-data.js:1109]: WDYM? we saw it returns like the new one in the bug, we saw the other way on the stream we had on AWS event

File: src/k-provider/ovp/external-captions-builder.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
 // @flow
 
+import OVPProviderParser from './provider-parser';
```
Comment (OrenMe) [inline src/k-provider/ovp/external-captions-builder.js:3]: this imports the entire class, if this is a util move it to a standalone export and import just that

File: test/src/k-provider/ovp/be-data.js [hunk 5]
```diff
@@ -2922,7 +2922,7 @@ const EntryWithBumperWithKs = {
       playbackCaptions: [
         {
           label: 'En',
-          format: '1',
+          format: '4',
```
Comment (OrenMe) [inline test/src/k-provider/ovp/be-data.js:2925]: is this change needed?

**Review by OrenMe — CHANGES_REQUESTED** [review]

File: test/src/k-provider/ovp/be-data.js (hunk 5 above)
Comment (Yuvalke) [reply test/src/k-provider/ovp/be-data.js:2925]: Yes, I want the parser to use VTT caption in this case to check query param and we're taking the VTT for format 2 and 4.

**Review by OrenMe — APPROVED** [review]
